### PR TITLE
chore: update gradle plugin version

### PR DESCRIPTION
**Pull request by naman14**
Update the Android gradle plugin to 3.0.0

Please Add Screenshots If any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ private boolean isCredentialsValid(final String username, final String password)
 
         final Resources resources = context.getResources();
         final String correctUsername = username.replaceFirst("\\s++$", "").trim();
-        if (username == null || username.trim().isEmpty()) {
+        if (username == null || username.matches("\\s*") || username.isEmpty()) {
             getMvpView().showUsernameError(context.getString(R.string.error_validation_blank,
```
Comment (naman14) [inline app/src/main/java/org/mifos/selfserviceapp/presenters/LoginPresenter.java:176]: PMD added [InefficientEmptyStringCheck](https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/strings.html#InefficientEmptyStringCheck) in PMD 3.6 which was updated with the gradle plugin update.

**Comment by therajanmaurya** [comment]
@naman14 I want one more change here, Please remove this from App Level to Project level depedencies 
```
buildscript {
    repositories {
        maven { url 'https://maven.fabric.io/public' }
    }

    dependencies {
        classpath 'io.fabric.tools:gradle:1.+'
    }
repositories {
    maven { url 'https://maven.fabric.io/public' }
}
}
```

and Please update the DBFlow to 4.x and after updating the DBFlow you will need to remove  `@ ModelContainer` (https://github.com/openMF/self-service-app/blob/development/app/src/main/java/org/mifos/selfserviceapp/models/Charge.java#L22) It's not available in 4.x. 

The last thing is to update the Dagger version 2.2 to 2.5.

**Comment by naman14** [comment]
@therajanmaurya updated the commit, please check

**Comment by therajanmaurya** [comment]
@naman14 remove this https://github.com/naman14/self-service-app/blob/0716493bb0cc7faa7e422b39b0be18a3f8078493/app/build.gradle#L4 too in the Project level

**Comment by naman14** [comment]
@therajanmaurya updated

**Pull request merged**